src/Facebook/FacebookBatchRequest.php

```diff
@@ -189,8 +191,19 @@ public function convertRequestsToJson()
     {
         $requests = [];
         foreach ($this->requests as $request) {
-            $attachedFiles = isset($request['attached_files']) ? $request['attached_files'] : null;
-            $requests[] = $this->requestEntityToBatchArray($request['request'], $request['name'], $attachedFiles);
+            $options = [];
+
+            if (!empty($request['name'])) {
+                $options['name'] = $request['name'];
+            }
+
+            if (!empty($request['attached_files'])) {
```

Comment:
BTW, you can change this too to `null !== $request['attached_files']` as `extractFileAttachments` always returns a string or null